Source_Code.txt

```diff
@@ -0,0 +1,50 @@
+def makePowerup():
+  pic=makePicture(pickAFile())
+  #getting the picture height helps alignment
+  picHeight=getHeight(pic)
+  
+  #the canvas is square so the width is the same as the height
+  canvasHeight=500
+  canvas=makeEmptyPicture(canvasHeight,canvasHeight, white)
+ 
+  show(canvas)
+  
+  #make pic centered
+  targetX=(canvasHeight-picHeight)/2
+  targetY=targetX
+
+  #number of times to move up and down
+  number=4
+
+  #make image move up then down
+  for loop in range (0,number):
+    #number of pixels move
+    maxHeight=10
+
+    #move image up
+    for up in range (0,maxHeight):
+      setAllPixelsToAColor(canvas, white)
+      copyInto(pic, canvas, targetX, targetY-up)
```

Comment:
This seems like a good approach --- however, you could grab the pixel, manipulate it, and place it in the new canvas within the body of a single loop.